### ISSUE TITLE
Gradle Module Metadata - publish by default

### ISSUE THREAD

**Issue by jjohannes** [gradle/gradle#10190]
Publish Gradle Module Metadata by default, which means enabling all behavior that is now hidden behind the [GRADLE_METADATA](https://github.com/gradle/gradle/blob/b39e588d94b67e54c0c8f789423e441b3b591d2d/subprojects/core/src/main/java/org/gradle/api/internal/FeaturePreviews.java#L30) preview flag. The flag should then be made inactive.

**Comment by jjohannes** [comment]
@ljacomet @melix I have merged #10274 already. Please have a look at the PR for more details and let me know if you have any comments.

I assume that this change will break some builds that work with custom repositories without proper metadata with a message like this:
```
> Could not find org:foo:1.0.
     Searched in the following locations:
       - https://repo.myrepo.org/repo/org/foo/1.0/foo-1.0.pom
```
I was wondering if we should extend the message with something like:
```
If the artifact you are trying to retrieve is located at the given position without metadata in 'pom' format, you need to adjust the 'metadataSources {}' (link to docs) of the 'https://repo.myrepo.org/repo/' repository declaration.
```

**Comment by jjohannes** [comment]
Looking into aritfact and classifier handling (traditional vs. GMM-based), I noticed a few more inconsistencies and found a small bug resulting in these changes:
- Do not drop variant attributes in results based on maven artifacts  389c12fa0f9743bd282b315ff7874c631c823807
- Align implementations of artifact identifier display names dbe50145bc78574cafe88c09fd49a41463485b87
- Fix AttributeConfigurationSelector to return the filtered result 3255f1d79b07db996f5ecd659013675c4e7e7028
